main/tests.py

```diff
@@ -1 +1,84 @@
-# Create your tests here.
+import pytest
+from dogowner.models import DogOwner
+
+
+@pytest.fixture
+def create_dog_owner_user():
+    return DogOwner.create(email='dogowner@address.com',
+                           username='dogOwnerUser01',
+                           password='password123',
+                           dog_name='dog name',
+                           first_name='test',
+                           last_name='user',
+                           phone_number=1234567890,
+                           dog_race='dog race',
+                           dog_picture_url='https://www.google.com/user1.jpg',
+                           dog_age=4,
+                           dog_weight=2,
+                           dog_gender='M'
+                           )
+
+
+@pytest.mark.django_db
+class TestLoginView:
+
+    def test_enter_login_page(self, client):
+        response = client.get('/login/')
+        assert response.status_code == 200
+
+    def test_valid_login_dog_owner_user_info(self, client, create_dog_owner_user):
+        prev_logged_user = client.get('/').wsgi_request.user
+        form = {'username': 'dogOwnerUser01',
+                'password': 'password123',
+                }
+
+        response = client.post('/login/', form, follow=True)
+        curr_log_user = response.wsgi_request.user
+        assert curr_log_user == create_dog_owner_user.user
+        assert prev_logged_user != curr_log_user
+
+    def test_invalid_login_dog_owner_user_info(self, client, create_dog_owner_user):
+        forms = {'username': "daycare@address.com",
+                 'password': "incorrect",
+                 }
+        response = client.post('/login/', forms, follow=True)
+        assert create_dog_owner_user.user != response.wsgi_request.user
+        assert b"Please enter a correct username and password." in response.content
```

Comment:
Asserting error messages is not a good practice and should be avoided when possible,
the reason for that is the fact that if the component that raises it changes you also need to change your tests.

Is there a way to assert that there was an error w/o asserting the error message?

Reply:
I think in that case just to check the response.wsgi_request.user user is not the user given in the form .